### PR TITLE
[RAPPS] Show the user that there are no search results

### PR DESCRIPTION
**Pull request by learn-more**
JIRA issue: [CORE-17944](https://jira.reactos.org/browse/CORE-17944)

### CODE REVIEW

File: base/applications/rapps/appview.cpp [hunk 1]
```diff
@@ -1127,6 +1127,30 @@ CAppsListView::~CAppsListView()
...
+    }
+    return lRes;
+}
+
+VOID CAppsListView::SetWatermark(const CStringW& Text)
+{
+    m_Watermark = Text;
+}
+
+
```
Comment (HBelusca) [inline base/applications/rapps/appview.cpp:1153]:
```suggestion
```

File: base/applications/rapps/appview.cpp [hunk 2]
```diff
@@ -1279,6 +1303,12 @@ HWND CAppsListView::Create(HWND hwndParent)
     SetImageList(m_hImageListView, LVSIL_SMALL);
     SetImageList(m_hImageListView, LVSIL_NORMAL);
 
+#pragma push_macro("SubclassWindow")
+#undef SubclassWindow
+    m_hWnd = NULL;
+    SubclassWindow(hwnd);
+#pragma pop_macro("SubclassWindow")
```
Comment (HBelusca) [inline base/applications/rapps/appview.cpp:1310]: What's happening there? Is it colliding with ATL definition?

File: base/applications/rapps/include/appview.h [hunk 3]
```diff
@@ -201,7 +201,7 @@ class CAppInfoDisplay :
 };
 
 class CAppsListView :
-    public CUiWindow<CListView>
+    public CUiWindow<CWindowImpl<CAppsListView, CListView>>
```
Comment (HBelusca) [inline base/applications/rapps/include/appview.h:204]: What's the reason of this change? Is it because CListView doesn't provide the direct way to modify the message map?

File: base/applications/rapps/gui.cpp [hunk 4]
```diff
@@ -706,6 +706,14 @@ VOID CMainWindow::UpdateApplicationsList(INT EnumType)
     m_ApplicationView->SetRedraw(TRUE);
     m_ApplicationView->RedrawWindow(0, 0, RDW_INVALIDATE | RDW_ALLCHILDREN); // force the child window to repaint
     UpdateStatusBarText();
+
+    CStringW text;
+    if (m_ApplicationView->GetItemCount() == 0 && !szSearchPattern.IsEmpty())
+    {
+        text.LoadString(IDS_NO_SEARCH_RESULTS);
+    }
+    m_ApplicationView->SetWatermark(text);
```
Comment (binarymaster) [inline base/applications/rapps/gui.cpp:715]: Just an idea... maybe improve it by including the search query string, like `No search results for "xxx"`

File: base/applications/rapps/appview.cpp (hunk 2 above)
Comment (learn-more) [reply base/applications/rapps/appview.cpp:1310]: `#define SubclassWindow(hwnd,lpfn) ((WNDPROC)SetWindowLongPtr((hwnd),GWLP_WNDPROC,(LPARAM)(WNDPROC)(lpfn)))`

File: base/applications/rapps/include/appview.h (hunk 3 above)
Comment (learn-more) [reply base/applications/rapps/include/appview.h:204]: CListView is a CWindow, which is just a wrapper for an HWND, meaning no message handling at all.

File: base/applications/rapps/gui.cpp (hunk 4 above)
Comment (learn-more) [reply base/applications/rapps/gui.cpp:715]: I think that's added complexity that does not serve a good purpose.

**Review by HBelusca — APPROVED** [review]
